### PR TITLE
Add BlockEncode support to Catalyst

### PR DESCRIPTION
**Pull request by dime10**
Normally this would be a CI only change to fetch the version of lightning that introduced support. But because we don't yet honor the `matrix` field in device toml files, `BlockEncode` also needs to be added to a "matrix" list in the frontend as a temporary measure.

### CODE REVIEW

File: frontend/catalyst/pennylane_extensions.py [hunk 1]
```diff
@@ -236,7 +236,7 @@ def _get_operations_to_convert_to_matrix(_config):
         # However, that affects differentiability.
         # None of the "matrix" gates with more than 2 qubits parameters are differentiable.
         # TODO: https://github.com/PennyLaneAI/catalyst/issues/398
-        return {"MultiControlledX"}
+        return {"MultiControlledX", "BlockEncode"}
```
[the remark this answers is not in the document]
Comment (maliasadi) [reply frontend/catalyst/pennylane_extensions.py:239]: Line 234, `MultiControlledX` is only mentioned.. this workaround is now for both `MultiControlledX` and `BlockEncode`, I assume

[the remark this answers is not in the document]
Comment (dime10) [reply frontend/catalyst/pennylane_extensions.py:239]: Ah got it, wasn't visible in the code snippet you targeted 👍

[the remark this answers is not in the document]
Comment (dime10) [reply frontend/catalyst/pennylane_extensions.py:239]: yeah :/

File: .github/workflows/build-wheel-linux-x86_64.yaml [hunk 2]
```diff
@@ -278,7 +278,7 @@ jobs:
               -DPYTHON_EXECUTABLE=$(which python${{ matrix.python_version }}) \
               -Dpybind11_DIR=$(python${{ matrix.python_version }} -c "import pybind11; print(pybind11.get_cmake_dir())") \
               -DENABLE_LIGHTNING_KOKKOS=ON \
-              -DLIGHTNING_GIT_TAG="v0.34.0" \
+              -DLIGHTNING_GIT_TAG="2716864521c539429bd382b92151a918d1a076ce" \
```
[the remark this answers is not in the document]
Comment (dime10) [reply .github/workflows/build-wheel-linux-x86_64.yaml:281]: Sounds good, I wasn't sure how important the full hash is, and I figured I can't hurt 🤷

[the remark this answers is not in the document]
Comment (dime10) [reply .github/workflows/build-wheel-linux-x86_64.yaml:281]: I think at this point we can simply consider updating the lightning git tag as part of the regular release process, but I'll put it in here for good measure.

File: frontend/catalyst/pennylane_extensions.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dime10) [reply frontend/catalyst/pennylane_extensions.py:239]: https://github.com/PennyLaneAI/catalyst/pull/483/commits/69c2d09345d38782865629339bf41f941fad4a29

File: doc/changelog.md [hunk 3]
```diff
@@ -4,6 +4,9 @@
 
 <h3>Improvements</h3>
 
+* Add support for the BlockEncode operator in Catalyst.
```
[the remark this answers is not in the document]
Comment (dime10) [reply doc/changelog.md]: https://github.com/PennyLaneAI/catalyst/pull/483/commits/69c2d09345d38782865629339bf41f941fad4a29

File: runtime/Makefile [hunk 4]
```diff
@@ -14,7 +14,7 @@ ENABLE_LIGHTNING?=ON
 ENABLE_LIGHTNING_KOKKOS?=OFF
 ENABLE_OPENQASM?=OFF
 ENABLE_ASAN?=OFF
-LIGHTNING_GIT_TAG_VALUE?="v0.34.0"
+LIGHTNING_GIT_TAG_VALUE?="2716864521c539429bd382b92151a918d1a076ce"
```
[the remark this answers is not in the document]
Comment (dime10) [reply runtime/Makefile:17]: https://github.com/PennyLaneAI/catalyst/pull/483/commits/69c2d09345d38782865629339bf41f941fad4a29

File: .github/workflows/build-wheel-linux-x86_64.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (dime10) [reply .github/workflows/build-wheel-linux-x86_64.yaml:282]: https://github.com/PennyLaneAI/catalyst/pull/483/commits/69c2d09345d38782865629339bf41f941fad4a29

**Review by maliasadi — APPROVED** [review]
:parachute:

**Pull request merged**